opentracing/src/main/scala/zio/telemetry/opentracing/OpenTracing.scala

```diff
@@ -26,54 +26,95 @@ object OpenTracing {
   val noop: URLayer[Clock, OpenTracing] =
     live(NoopTracerFactory.create())
 
+  /**
+   * Creates a OpenTracing layer which will be constructed each time it's provided by provideLayer (fresh)
+   *
+   * On each construction the current value of the `FiberRef[A]` is extracted to a `Entrypoint` which contains
+   * a name plus a SpanContext. Therefore if you use `FiberRef.locally` and provide a local context and inside
+   * locally you use `provideLayer` the OpenTracing layer will be scoped to that `Entrypoint`.
+   *
+   * This can be plugged at points in your application where messages come in from a async boundary like a http server,
+   * kafka consumer, etc.
+   *
+   * @param tracer A instance of a OpenTracing tracer
+   * @param refEntrypoint A FiberRef with the context
+   * @param entrypointExtractor A `EntrypointExtractor` which extracts a `Entrypoint` from `A`
+   * @tparam A The generic context
+   * @return OpenTracing layer
+   */
+  def fromEntrypoint[A](
+    tracer: Tracer,
+    refEntrypoint: FiberRef[A],
+    entrypointExtractor: EntrypointExtractor[A]
+  ): URLayer[Clock, OpenTracing] =
+    ZLayer.fromManaged(managedFromEntrypoint(tracer, refEntrypoint, entrypointExtractor)).fresh
+
   def live(tracer: Tracer, rootOperation: String = "ROOT"): URLayer[Clock, OpenTracing] =
     ZLayer.fromManaged(managed(tracer, rootOperation))
 
+  def managedFromEntrypoint[A](
+    tracer0: Tracer,
+    refEntrypoint: FiberRef[A],
```

Comment:
I'm not sure I understand why we need to provide `refEntrypoint` here. Can't we just create a fresh one inside `managedFromEntrypoint`? Ha ving to provide a `FiberRef` feels a bit low level.

Reply:
The idea is to create the layer with an external mutually variable, hence a `FiberRef`. Upon creation it will extract the current context which set typically by calling `.locally`. This happens when you process a HTTP request or a Kafka message.